app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -65,12 +65,13 @@ def dest_folder
     host_dc = dest_host.parent_datacenter || dest_host.ems_cluster.parent_datacenter
 
     # Pick 'Discovered virtual machine' or its parent folder in the destination datacenter
-    vm_folder = "Datacenters/#{host_dc.name}/vm"
+    vm_folder = "#{host_dc.folder_path}/vm"
     find_folder("#{vm_folder}/Discovered virtual machine", host_dc) || find_folder(vm_folder, host_dc)
   end
 
   def find_folder(folder_path, datacenter)
-    EmsFolder.where(:name => File.basename(folder_path)).detect do |f|
+    manager_id = source.ext_management_system.id
+    EmsFolder.where(:name => File.basename(folder_path), :ems_id => manager_id).detect do |f|
       f.folder_path == folder_path && f.parent_datacenter == datacenter
     end
```

Comment:
Man this is expensive :(  We should probably clean this up in another way in the future.